src/workloads/scale/BulkLoading.yml

```diff
@@ -0,0 +1,287 @@
+
+# 4 Load
+# 5 Index
+# 6 Measure - BulkWriteSingleConnectionWithIndex
+# 7 Delete
+# 8 Quiese
+
+# 9 Load
+# 10 Measure - BulkWriteManyConnectionsNoIndex
+# 11 Delete
+# 12 Quiesce
+
+# 13 Load
+# 14 Index
+# 15 Measure - BulkWriteManyConnectionsWithIndex
+# 16 Delete
+# 17 Quiese
+
+# 18 Load
+# 19 Measure - NormalWriteManyConnectionsNoIndex
+# 20 Delete
+# 21 Quiese
+
+# 22 Load
+# 23 Index
+# 24 Measure - NormalWriteManyConnectionsWithIndex
+# 25 Delete
+# 26 Quiese
+
+# 27 Load
+# 28 Index
+# 29 Measure - BulkUpsertsManyConnectionsWithIndex
+# 30 Delete
+# 31 Quiese
+# ---End Phase Description---
+
+Clients:
+  Default:
+    QueryOptions:
+      socketTimeoutMS: -1
+      maxPoolSize: 400
+
+GlobalDefaults:
+  - &NumPhases 31
+  - &DbName db
+  - &CollectionName coll
+
+  # This document layout leads to ~650% CPU utilization on the genny client,
+  # so this is close to the maximum document complexity we can use here.
+  # Using ^RandomDate for the timestamp is too compute intensive.
+  - &Document {
+      sensorId: { ^RandomInt: { min: 0, max: 200000000 } },
+      timestamp: { ^RandomInt: { min: 1577836800000, max: 1692020604000 } },
+      eventId: { ^RandomInt: { min: 1, max: 10 } },
+      dataField1: { ^FastRandomString: { length: 10 } },
+      dataField2: { ^FastRandomString: { length: 20 } },
+      dataField3: { ^FastRandomString: { length: 30 } },
+      dataField4: { ^FastRandomString: { length: 40 } },
+      dataField5: { ^FastRandomString: { length: 50 } },
+    }
+
+  # This workload uses a batch size of 100 for the bulk writes.
+  # We evaluated the impact of different batch sizes and the document throughput
+  # does not appear to be impacted by the batch size.
+  - &WriteOp { WriteCommand: insertOne, Document: *Document }
+  - &WriteOps10x [
+      *WriteOp, *WriteOp, *WriteOp, *WriteOp, *WriteOp,
+      *WriteOp, *WriteOp, *WriteOp, *WriteOp, *WriteOp,
+    ]
+  - &WriteOps100x {
+      ^FlattenOnce:
+        [
+          *WriteOps10x, *WriteOps10x, *WriteOps10x, *WriteOps10x, *WriteOps10x,
+          *WriteOps10x, *WriteOps10x, *WriteOps10x, *WriteOps10x, *WriteOps10x,
+        ],
+    }
+  - &UpsertOp {
+      WriteCommand: updateOne,
+      Filter: { sensorId: { ^RandomInt: { min: 0, max: 200000000 } } },
+      Update:
+        {
+          $set:
+            { eventId: 0, dataField1: { ^FastRandomString: { length: 10 } } },
+        },
+      OperationOptions: { Upsert: true },
+    }
+  - &UpsertOps10x [
+      *UpsertOp, *UpsertOp, *UpsertOp, *UpsertOp, *UpsertOp,
+      *UpsertOp, *UpsertOp, *UpsertOp, *UpsertOp, *UpsertOp,
+    ]
+  - &UpsertOps100x {
+      ^FlattenOnce:
+        [
+          *UpsertOps10x, *UpsertOps10x, *UpsertOps10x, *UpsertOps10x, *UpsertOps10x,
+          *UpsertOps10x, *UpsertOps10x, *UpsertOps10x, *UpsertOps10x, *UpsertOps10x,
+        ],
+    }
+
+ActorTemplates:
+  - TemplateName: BulkWriterTemplate
+    Config:
+      Name: { ^Parameter: { Name: "Name", Default: "unset" } }
```

Comment:
hmm should the 'unset' values just have a reasonable default instead?

Reply:
This did make the file a bit shorter :)

Reply:
What happens to the remaining 'unsets'. I'd almost prefer a 'Mandatory: true' / 'Required: true' option where genny throws an exception if there is no value.

Maybe this is a genny feature request?

Reply:
I'll open a Genny ticket for this, I've seen a lot of workloads coming up with custom indicators that a parameter does not have a valid default. My dream behavior here would be that the default parameter is optional and that genny errors out if the parameter is not provided and there is no default.

Reply:
Opened up [EVG-20735](https://jira.mongodb.org/browse/EVG-20735)

Reply:
:+1: